src/display-comments.js

```diff
@@ -0,0 +1,145 @@
+</form>
+</div>
+</div>
+<span class="close-modul"><ion-icon name="close-outline"></ion-icon></span>
+</div>
+`;
+
+  document.body.insertAdjacentHTML('afterbegin', htmlText);
+};
+
+const displayIngredients = (ingredients, container) => {
+  ingredients.forEach((ingr) => {
+    const textHTML = `
+<li class="ingredient">${ingr.quantity === null ? '' : ingr.quantity} ${
+  ingr.unit === 'tsp' ? 'teaspoons' : ingr.unit
+} ${ingr.unit === 'tsp' || ingr.unit === 'cup' ? 'of' : ''} ${
+  ingr.description
+}</li>`;
+    container.insertAdjacentHTML('beforeend', textHTML);
+  });
+};
+
+const closePopUp = (x, popUp) => {
+  x.addEventListener('click', () => {
+    popUp.style.display = 'none';
+  });
+};
+
+const commentCounter = (arr) => {
+  const newArr = [...arr];
+  const arrLenght = newArr.length;
+  return arrLenght;
+};
+
+const commentCounterApi = async (value) => {
+  const commNum = await getComm(value);
+  if (commNum) {
+    return commentCounter(commNum);
+  }
+  return null;
+};
+
+const getCommArray = async (pizzaId, container) => {
+  const commArr = await getComm(pizzaId);
+  if (commArr) {
+    container.style.display = 'flex';
+    commArr.forEach((comm) => {
+      const textHTML = `
+<li class="comment">${comm.creation_date}
+<span class="user-name">${comm.username}</span>: ${comm.comment}
+</li>
+`;
+      container.insertAdjacentHTML('afterbegin', textHTML);
+    });
+  }
+};
+
+const commCounterDisplay = async (value, container) => {
+  const newValue = await value;
+  if (!newValue) container.textContent = 'No comments yet';
+  if (newValue === 1) container.textContent = '1 Comment';
+  if (newValue > 1) container.textContent = `Comments (${newValue})`;
+};
+
+const buildPopUp = async (e) => {
+  const pizzaId = e.target.id;
+  const apiResult = await getRecipe(pizzaId);
+
+  domManip(apiResult);
+
+  const ingredientsList = document.querySelector('.ingredients-list');
+  const closeIcon = document.querySelector('.close-modul');
+  const popUpDiv = document.querySelector('.pop-up-comment-div');
+  const form = document.querySelector('.leave-comm');
+  const inputName = document.querySelector('.input-name');
+  const inputMess = document.querySelector('.input-mess');
+  const commentsList = document.querySelector('.comments-list');
+  const comments = document.querySelector('.comments-title');
+
+  popUpDiv.style.display = 'block';
+
+  displayIngredients(apiResult.ingredients, ingredientsList);
+
+  getCommArray(pizzaId, commentsList);
+
+  commCounterDisplay(commentCounterApi(pizzaId), comments);
+
+  form.addEventListener('submit', (e) => {
+    e.preventDefault();
+    sendNewComm(pizzaId, inputName.value, inputMess.value);
+    inputName.value = '';
+    inputMess.value = '';
+    setTimeout(() => {
+      commentsList.innerHTML = '';
+      getCommArray(pizzaId, commentsList);
+      commCounterDisplay(commentCounterApi(pizzaId), comments);
+    }, 500);
+  });
```

Comment:
- [ ] OPTIONAL: Everything looks great, but it seems that when a user submits a comment, the comment shows up only after a refresh. Kindly work on this in your spare time. :pray:

Reply:
**Thanks a lot for the feedback.**
It seems that half a second does not work all the time, so I change it to 1 sec, so now never miss a new comment